src/coreclr/vm/arm64/asmhelpers.S

```diff
@@ -231,12 +232,28 @@ LOCAL_LABEL(EphemeralCheckEnabled):
     PREPARE_EXTERNAL_VAR g_highest_address, x12
     ldr  x6, [x12]
 
+    PREPARE_EXTERNAL_VAR g_region_to_generation_table, x12
+    ldr  x7, [x12]
+
+    PREPARE_EXTERNAL_VAR g_region_shr, x12
+    ldr  w10, [x12]
+
+    PREPARE_EXTERNAL_VAR g_region_use_bitwise_write_barrier, x12
+    ldr  w11, [x12]
+
+    // Only allow bitwise write barriers if LSE atomics are present
+    PREPARE_EXTERNAL_VAR g_arm64_atomics_present, x12
```

Comment:
This check should be done when initializing `g_region_use_bitwise_write_barrier`

Reply:
Done.